test/1_stdlib/StringDiagnostics.swift

```diff
@@ -60,19 +60,30 @@ func testStringDeprecation(hello: String) {
 }
 
 // Positive and negative tests for String index types
-func acceptsForwardIndex<I: ForwardIndex>(index: I) {}
-func acceptsBidirectionalIndex<I: BidirectionalIndex>(index: I) {}
-func acceptsRandomAccessIndex<I: RandomAccessIndex>(index: I) {}
+func acceptsCollectionIndex<I: Comparable>(index: I) {}
+func acceptsBidirectionalCollectionIndex<I: Comparable>(index: I) {}
+func acceptsRandomAccessCollectionIndex<I: Strideable>(index: I) {}
 
 func testStringIndexTypes(s: String) {
-  acceptsForwardIndex(s.utf8.startIndex)
-  acceptsBidirectionalIndex(s.utf8.startIndex) // expected-error{{argument type 'String.UTF8View.Index' does not conform to expected type 'BidirectionalIndex'}}
-  acceptsBidirectionalIndex(s.unicodeScalars.startIndex)
-  acceptsRandomAccessIndex(s.unicodeScalars.startIndex) // expected-error{{argument type 'String.UnicodeScalarView.Index' does not conform to expected type 'RandomAccessIndex'}}
-  acceptsBidirectionalIndex(s.characters.startIndex)
-  acceptsRandomAccessIndex(s.characters.startIndex) // expected-error{{argument type 'String.CharacterView.Index' does not conform to expected type 'RandomAccessIndex'}}
-  
+  acceptsCollectionIndex(s.utf8.startIndex)
+  acceptsBidirectionalCollectionIndex(s.utf8.startIndex)
+  acceptsRandomAccessCollectionIndex(s.utf8.startIndex) // expected-error{{argument type 'String.UTF8View.Index' does not conform to expected type 'Strideable'}}
+
   // UTF16View.Index is random-access with Foundation, bidirectional without
-  acceptsRandomAccessIndex(s.utf16.startIndex)
+  acceptsCollectionIndex(s.utf16.startIndex)
+  acceptsBidirectionalCollectionIndex(s.utf16.startIndex)
+  acceptsRandomAccessCollectionIndex(s.utf16.startIndex)
```

Comment:
Should I add XxxView collection conformance and also keep the XxxView.index conformance tests? I guess you can't screw up the later if you have the former correct... will change.